From b73227e97086116e596206b22ce0356bfc9b0a2c Mon Sep 17 00:00:00 2001
From: Awais Belal <awais_belal@mentor.com>
Date: Fri, 11 Nov 2016 14:48:54 +0500
Subject: [PATCH] vulkaninfo.c: fix segfault when DISPLAY is not set

Both xlib and xcb interfaces expect the DISPLAY environment
variable to be set before creation of a window and the
display creation mechanism would segfault if that is
not the case and won't provide the user with details on
what has to be done to correct the problem.
We now handle such scenarios and exit cleanly after
providing the user with some details.

Signed-off-by: Awais Belal <awais_belal@mentor.com>
---
 demos/vulkaninfo.c | 14 ++++++++++++++
 1 file changed, 14 insertions(+)

Index: git/demos/vulkaninfo.c
===================================================================
--- git.orig/demos/vulkaninfo.c	2017-02-22 14:43:03.752917439 -0600
+++ git/demos/vulkaninfo.c	2017-02-22 14:56:57.000000000 -0600
@@ -936,6 +936,9 @@
         exit(1);
     }
 
+    if (inst->xlib_display == NULL)
+        return;
+
     XVisualInfo vInfoTemplate={};
     vInfoTemplate.screen = DefaultScreen(inst->xlib_display);
     XVisualInfo *visualInfo = XGetVisualInfo(inst->xlib_display, visualMask,
@@ -1570,6 +1573,10 @@
     if (HasExtension(VK_KHR_XCB_SURFACE_EXTENSION_NAME,
                       inst.global_extension_count, inst.global_extensions)) {
         AppCreateXcbWindow(&inst);
+        if (inst.xlib_display == NULL) {
+            printf("'DISPLAY' variable not set correctly. Exiting!\n'");
+            goto out;
+        }
         for (i = 0; i < gpu_count; i++) {
             AppCreateXcbSurface(&inst);
             printf("GPU id       : %u (%s)\n", i, gpus[i].props.deviceName);
@@ -1606,6 +1613,7 @@
         printf("\n\n");
     }
 
+out:
     for (i = 0; i < gpu_count; i++)
         AppGpuDestroy(&gpus[i]);
     free(gpus);
